cli/src/lib.rs

```diff
@@ -3844,8 +3844,9 @@ fn serialize_idl(idl: &Idl) -> Result<Vec<u8>> {
 }
 
 fn serialize_idl_ix(ix_inner: anchor_lang::idl::IdlInstruction) -> Result<Vec<u8>> {
-    let mut data = anchor_lang::idl::IDL_IX_TAG.to_le_bytes().to_vec();
-    data.append(&mut ix_inner.try_to_vec()?);
+    let mut data = Vec::with_capacity(256);
+    data.extend_from_slice(&anchor_lang::idl::IDL_IX_TAG.to_le_bytes());
+    ix_inner.serialize(&mut data)?;
```

Comment:
We serialize 1000 bytes per each IDL `Write` instruction(most IDLs are bigger):

https://github.com/coral-xyz/anchor/blob/4c0af6dc5394c714a70296eae686b261f520f206/cli/src/lib.rs#L2397

CLI and client changes are negligible but I'm onboard with including them for the sake of consistency.